### ISSUE TITLE
Unable to use reactive and jdbc datasource simultaneously with Hibernate reactive

### ISSUE THREAD

**Issue by priyanknarvekar** [quarkusio/quarkus#35485]
### Describe the bug

if two data sources are added 
- one based on a inbuild reactive driver, 
- and another based on jdbc driver
And trying to leverage Hibenate Reactive  throws an error and complains `The datasource 'db1' is not configured but the persistence unit 'db1' uses it.`, however it is indeed configured but is just a reactive datasource and not jdbc

Looks like hibernate tries to look for jdbc datasource, even if the corresponding datasource is configured as reactive

example of the configuration used 
```
quarkus.datasource."db1".db-kind=postgresql 
quarkus.datasource."db1".reactive.url=vertx-reactive:postgresql://localhost:5432/db1
quarkus.datasource."db1".username=hibernate
quarkus.datasource."db1".password=hibernate
quarkus.datasource."db1".jdbc=false
quarkus.datasource."db1".reactive=true
quarkus.datasource."db1".reactive.max-size=3

quarkus.hibernate-orm."db1".datasource=db1
quarkus.hibernate-orm."db1".packages=com.example.models
quarkus.hibernate-orm."db1".jdbc.statement-batch-size=1000
quarkus.hibernate-orm."db1".database.generation=none
quarkus.hibernate-orm."db1".database.generation.create-schemas=false
quarkus.hibernate-orm."db1".database.default-schema=dev
quarkus.hibernate-orm."db1".log.sql=true


quarkus.datasource."db2".db-kind=postgresql
quarkus.datasource."db2".jdbc.url=jdbc:postgresql://localhost:5433/db2
quarkus.datasource."db2".username=hibernate
quarkus.datasource."db2".password=hibernate
quarkus.datasource."db2".jdbc=true
quarkus.datasource."db2".reactive=false
quarkus.datasource."db2".jdbc.max-size=3
```
https://github.com/priyanknarvekar/datasource-demo/blob/main/src/main/resources/application.properties

### Expected behavior

Application Starts, and hibernate reactive correctly looks for the reactive datasource

### Actual behavior

Throws an error and quarkus fails to start:
```
Listening for transport dt_socket at address: 5005
2023-08-23 06:32:40,656 WARN  [io.qua.dat.dep.dev.DevServicesDatasourceProcessor] (build-23) Unable to determine a database type for default datasource
2023-08-23 06:32:40,659 WARN  [io.qua.hib.orm.dep.HibernateOrmProcessor] (build-11) Could not find a suitable persistence unit for model classes:
        - io.quarkus.hibernate.reactive.panache.PanacheEntity
        - io.quarkus.hibernate.reactive.panache.PanacheEntityBase
2023-08-23 06:32:40,689 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure
2023-08-23 06:32:41,200 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor#configurationDescriptorBuilding threw an exception: io.quarkus.runtime.configuration.ConfigurationException: The datasource 'db1' is not configured but the persistence unit 'db1' uses it. To solve this, configure datasource 'db1'. Refer to https://quarkus.io/guides/datasource for guidance.
        at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.lambda$findJdbcDataSource$13(HibernateOrmProcessor.java:1227)
        at java.base/java.util.Optional.orElseThrow(Optional.java:403)
        at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.findJdbcDataSource(HibernateOrmProcessor.java:1227)
        at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.producePersistenceUnitDescriptorFromConfig(HibernateOrmProcessor.java:910)
        at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.handleHibernateORMWithNoPersistenceXml(HibernateOrmProcessor.java:878)
        at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.configurationDescriptorBuilding(HibernateOrmProcessor.java:332)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:578)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:858)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
        at java.base/java.lang.Thread.run(Thread.java:1589)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)

        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:336)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:253)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:60)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:82)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:423)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:55)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:138)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:93)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor#configurationDescriptorBuilding threw an exception: io.quarkus.runtime.configuration.ConfigurationException: The datasource 'db1' is not configured but the persistence unit 'db1' uses it. To solve this, configure datasource 'db1'. Refer to https://quarkus.io/guides/datasource for guidance.
        at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.lambda$findJdbcDataSource$13(HibernateOrmProcessor.java:1227)
        at java.base/java.util.Optional.orElseThrow(Optional.java:403)
        at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.findJdbcDataSource(HibernateOrmProcessor.java:1227)
        at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.producePersistenceUnitDescriptorFromConfig(HibernateOrmProcessor.java:910)
        at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.handleHibernateORMWithNoPersistenceXml(HibernateOrmProcessor.java:878)
        at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.configurationDescriptorBuilding(HibernateOrmProcessor.java:332)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:578)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:858)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
        at java.base/java.lang.Thread.run(Thread.java:1589)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)

        at io.quarkus.builder.Execution.run(Execution.java:123)
        at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:160)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:332)
        ... 9 more
Caused by: io.quarkus.runtime.configuration.ConfigurationException: The datasource 'db1' is not configured but the persistence unit 'db1' uses it. To solve this, configure datasource 'db1'. Refer to https://quarkus.io/guides/datasource for guidance.
        at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.lambda$findJdbcDataSource$13(HibernateOrmProcessor.java:1227)
        at java.base/java.util.Optional.orElseThrow(Optional.java:403)
        at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.findJdbcDataSource(HibernateOrmProcessor.java:1227)
        at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.producePersistenceUnitDescriptorFromConfig(HibernateOrmProcessor.java:910)
        at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.handleHibernateORMWithNoPersistenceXml(HibernateOrmProcessor.java:878)
        at io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor.configurationDescriptorBuilding(HibernateOrmProcessor.java:332)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:578)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:858)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
        at java.base/java.lang.Thread.run(Thread.java:1589)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

### How to Reproduce?

Reproducer:
https://github.com/priyanknarvekar/datasource-demo/

Steps:
1. Start the database containers
```
docker run -d --name postgres-quarkus-hibernate-db1 -e POSTGRES_USER=hibernate \
           -e POSTGRES_PASSWORD=hibernate -e POSTGRES_DB=db1 \
           -p 5432:5432 postgres:14.1

docker run -d --name postgres-quarkus-hibernate-db2 -e POSTGRES_USER=hibernate \
           -e POSTGRES_PASSWORD=hibernate -e POSTGRES_DB=db2 \
           -p 5433:5432 postgres:14.1
```
2.  clone the reproducer - git@github.com:priyanknarvekar/datasource-demo.git
3. start the application `quarkus dev`

### Output of `uname -a` or `ver`

Darwin 19.6.0 Darwin Kernel Version 19.6.0: Tue Jun 21 21:18:39 PDT 2022; root:xnu-6153.141.66~1/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "19.0.2" 2023-01-17

### GraalVM version (if different from Java)

OpenJDK Runtime Environment GraalVM CE 22.3.1 (build 19.0.2+7-jvmci-22.3-b12)

### Quarkus version or git rev

QUARKUS 3.2.4

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

pache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63)

**Comment by geoand** [comment]
This is a duplicate of https://github.com/quarkusio/quarkus/issues/13425

**Comment by priyanknarvekar** [comment]
@geoand 
unless I am mistaken, difference from that issues is i am not trying to use non-reactive Hibernate, but only

- Reactive Datasouce with Reactive Hibernate ( db1 )
- Regular Datasource (for db2, do not need the regular orm for this)

there is not hibernate-orm dependency that was at least explicitly added , 
the project does not have `io.quarkus:quarkus-hibernate-orm` as dependency 
so there should not be a conflict, but when Reactive hibernate boots , should correctly use the reactive datasouce.

**Comment by geoand** [comment]
cc @gsmet @yrodiere

**Comment by yrodiere** [comment]
Hello,

TL;DR: there are documentation problems, but what you're trying to do isn't possible at the moment, see #33380

So, first, let me say right away that the error message and documentation definitely need improvement.

However, the thing is, I don't think the Hibernate Reactive extension currently supports named persistence unit.

The [documentation mentions](https://quarkus.io/guides/hibernate-reactive#hr-limitations):

> it’s not possible to configure multiple persistence units at the moment

It's not technically accurate though. From what I can see in the code, it's just that the Hibernate Reactive extension only supports the default (unnamed) persistence unit:

https://github.com/quarkusio/quarkus/blob/265bd2e97f4724a9eba45e9c05e7ed0eabcab0e9/extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveProcessor.java#L174-L194

What's more, and that's undocumented, it doesn't support using a datasource other than the default one:

https://github.com/quarkusio/quarkus/blob/265bd2e97f4724a9eba45e9c05e7ed0eabcab0e9/extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveProcessor.java#L160-L162

I'll send a PR right away to address the documentation issues, but I believe addressing the limitation will only happen when addressing #13425.

In your case, I suspect using this configuration instead should take you a step further:

```
# Default datasource for Hibernate Reactive
quarkus.datasource.db-kind=postgresql 
quarkus.datasource.reactive.url=vertx-reactive:postgresql://localhost:5432/db1
quarkus.datasource.username=hibernate
quarkus.datasource.password=hibernate
quarkus.datasource.jdbc=false
quarkus.datasource.reactive=true
quarkus.datasource.reactive.max-size=3

# Default persistence unit for Hibernate Reactive, using the default datasource
quarkus.hibernate-orm.packages=com.example.models
quarkus.hibernate-orm.jdbc.statement-batch-size=1000
quarkus.hibernate-orm.database.generation=none
quarkus.hibernate-orm.database.generation.create-schemas=false
quarkus.hibernate-orm..database.default-schema=dev
quarkus.hibernate-orm.log.sql=true

# Named datasource for direct JDBC access
quarkus.datasource."db2".db-kind=postgresql
quarkus.datasource."db2".jdbc.url=jdbc:postgresql://localhost:5433/db2
quarkus.datasource."db2".username=hibernate
quarkus.datasource."db2".password=hibernate
quarkus.datasource."db2".jdbc=true
quarkus.datasource."db2".reactive=false
quarkus.datasource."db2".jdbc.max-size=3
```

But then you will hit #33380, which someone attempted to solve in #33790, but that PR is currently stuck in limbo, waiting for someone to address the comments in the review...

To sum up, I'll address the documentation problems, but I think we should close this as a duplicate of #33380

**Comment by priyanknarvekar** [comment]
@yrodiere sounds good ! makes sense ! Thanks!

**Comment by mikadev** [comment]
why is not supported to mix both or having 2 reactive in the same project

**Comment by yrodiere** [comment]
Back when this thread was opened, the answer to this question was, as often, "nobody worked on this yet". Not sure what else you would expect.

Anyway, this thread is old. It _is_ supported now.

Mixing blocking and reactive Hibernate added in 3.22: https://github.com/quarkusio/quarkus/pull/44473
Multiple persistence units in Hibernate Reactive added in 3.26: https://github.com/quarkusio/quarkus/pull/48007

If you have further questions, please use stackoverflow or [GitHub discussions](https://github.com/quarkusio/quarkus/discussions).
